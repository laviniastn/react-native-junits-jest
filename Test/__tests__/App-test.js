/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

configure({adapter: new Adapter()});

describe('App Component', () => {

  it('App renders correctly', () => {
    const tree = renderer.create(
      <App />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('App initial state ', () => {
    const wrapper = shallow(<App />);

    const componentInstance = wrapper.instance();

    expect(wrapper.state('isLogined')).toEqual(false);
    expect(wrapper.state('username')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
 
  });

  it('App valid login', () => {
    const wrapper = shallow(<App />);
    const componentInstance = wrapper.instance();
    componentInstance.setState({username: 'lavinia', password: '123456'})

    componentInstance.login();

    expect(wrapper.state('username')).toEqual('lavinia');
    expect(wrapper.state('password')).toEqual('123456');
    expect(wrapper.state('isLogined')).toEqual(true);
    
  });

  it('App invalid login - wrong password', () => {
    const wrapper = shallow(<App />);
    const componentInstance = wrapper.instance();
    componentInstance.setState({username: 'lavinia', password: '12345'})

    componentInstance.login();

    expect(wrapper.state('username')).toEqual('lavinia');
    expect(wrapper.state('password')).toEqual('12345');
    expect(wrapper.state('isLogined')).toEqual(false);
    
  });

  it('App invalid login - wrong username', () => {
    const wrapper = shallow(<App />);
    const componentInstance = wrapper.instance();
    componentInstance.setState({username: 'lstan', password: '123456'})

    componentInstance.login();

    expect(wrapper.state('username')).toEqual('lstan');
    expect(wrapper.state('password')).toEqual('123456');
    expect(wrapper.state('isLogined')).toEqual(false);
    
  });

  it('App invalid login - null input', () => {
    const wrapper = shallow(<App />);
    const componentInstance = wrapper.instance();

    componentInstance.login();

    expect(wrapper.state('username')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
    expect(wrapper.state('isLogined')).toEqual(false);
    
  });
});



