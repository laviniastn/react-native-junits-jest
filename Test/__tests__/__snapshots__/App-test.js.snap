// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  accessibilityLabel="app-root"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="app-root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="username"
      name="username"
      onChangeText={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="password"
      name="password"
      onChangeText={[Function]}
      secureTextEntry={true}
    />
  </View>
  <Text
    accessibilityLabel="loginstatus"
  >
    fail
  </Text>
  <View
    accessibilityLabel="login"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00b5ec",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`App Component App renders correctly 1`] = `
<View
  accessibilityLabel="app-root"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="app-root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="username"
      name="username"
      onChangeText={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="password"
      name="password"
      onChangeText={[Function]}
      secureTextEntry={true}
    />
  </View>
  <Text
    accessibilityLabel="loginstatus"
  >
    fail
  </Text>
  <View
    accessibilityLabel="login"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00b5ec",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`App Component renders correctly 1`] = `
<View
  accessibilityLabel="app-root"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="app-root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="username"
      name="username"
      onChangeText={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="password"
      name="password"
      onChangeText={[Function]}
      secureTextEntry={true}
    />
  </View>
  <Text
    accessibilityLabel="loginstatus"
  >
    fail
  </Text>
  <View
    accessibilityLabel="login"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00b5ec",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  accessibilityLabel="app-root"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="app-root"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="username"
      name="username"
      onChangeText={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderBottomColor": "#F5FCFF",
        "borderBottomWidth": 1,
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1,
        "flexDirection": "row",
        "marginBottom": 20,
        "width": "80%",
      }
    }
  >
    <TextInput
      accessibilityLabel="password"
      name="password"
      onChangeText={[Function]}
      secureTextEntry={true}
    />
  </View>
  <Text
    accessibilityLabel="loginstatus"
  >
    fail
  </Text>
  <View
    accessibilityLabel="login"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00b5ec",
        "borderRadius": 30,
        "flexDirection": "row",
        "height": 45,
        "justifyContent": "center",
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
